Tools/ardupilotwaf/git_submodule.py

```diff
@@ -111,8 +112,62 @@ def u(s):
     def __str__(self):
         return 'Submodule update: %s' % self.submodule
 
+def check_folder(path_list=[]):
+    for d in Utils.to_list(path_list):
+        listdir = os.listdir(d)
+        for f in listdir(d):
+            if f.startsith('.'):
+                listdir.remove()
+        if len(f) == 0:
+            raise
+    return f
+
+def module_is_uninitialised(d):
+    listdir = os.listdir(d)
+    listdir_trimmed = [i for i in listdir if not i.startswith('.')]
+    print("listdir_trimmed", listdir_trimmed)            
+    if len(listdir_trimmed) == 0:
+        return True
+
+    return False
+
+
 def configure(cfg):
+    print("============CONFIGURE===================")
     cfg.find_program('git')
+    path = cfg.srcnode.abspath()+'/modules/'
+    check_modules = ['ChibiOS', 'gbenchmark', 'gtest', 'libcanard', 'mavlink', 'uavcan']
+    missing_modules = []
+
+    for i in check_modules:
+        path_module = (path+i)
+        print("CHECK MODULE", i)
+        if module_is_uninitialised(path_module) == True:
```

Comment:
```suggestion
        if module_is_uninitialised(path_module):
```

Isn't this inverted?